pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -213,20 +248,157 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
+	klog.InfoS("Updating WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport)
+	if ciImport.Spec.WireGuard == nil || ciImport.Spec.WireGuard.PublicKey == "" {
+		klog.InfoS("ClusterInfoImport's WireGuard field has not been initialized, skip it", "ClusterInfoImport", ciImport)
+		return nil
+	}
+	serviceIP, serviceCIDR, err := net.ParseCIDR(ciImport.Spec.ServiceCIDR)
+	if err != nil {
+		return err
+	}
+	remoteWireGuardIP := serviceIP.Mask(serviceCIDR.Mask)
+	_, remoteWireGuardNet, err := net.ParseCIDR(remoteWireGuardIP.String() + "/32")
+	if err != nil {
+		return err
+	}
+
+	gatewayIP := net.ParseIP(ciImport.Spec.GatewayInfos[0].GatewayIP)
+	allowedIPs := []*net.IPNet{remoteWireGuardNet}
+	if err := c.wireGuardClient.UpdatePeer(ciImport.Name, ciImport.Spec.WireGuard.PublicKey, gatewayIP, allowedIPs); err != nil {
+		klog.ErrorS(err, "Failed to update WireGuard peer", "ClusterInfoImport", ciImport)
+		return err
+	}
+	klog.InfoS("Updated WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport.Name)
+	if err := c.addWireGuardRoute(remoteWireGuardIP.String()); err != nil {
+		klog.ErrorS(err, "Failed to add WireGuard route on the host", "remoteGateway", ciImport.Name)
+		return err
+	}
+	return nil
+}
+
+// initializeWireGuard initializes the WireGuard interface and client.
+// It will also update Gateway's WireGuard field.
+func (c *MCDefaultRouteController) initializeWireGuard() error {
+	c.wireGuardConfig.MTU = c.nodeConfig.NodeTransportInterfaceMTU - c.networkConfig.MTUDeduction - config.WireGuardOverhead
+	wgClient, err := wireGuardNewFunc(c.nodeConfig, c.wireGuardConfig)
+	if err != nil {
+		return err
+	}
+	c.wireGuardClient = wgClient
+
+	clusterServiceCIDR, err := c.serviceCIDRProvider.GetServiceCIDR(false)
+	if err != nil {
+		return err
+	}
+	publicKey, err := c.wireGuardClient.Init(clusterServiceCIDR.IP.Mask(clusterServiceCIDR.Mask), nil)
+	if err != nil {
+		return err
+	}
+
+	patch, _ := json.Marshal(map[string]interface{}{
+		"wireGuard": map[string]interface{}{
+			multiclusterWireGuardPublicKey: publicKey,
+		},
+	})
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_, err := c.mcClient.MulticlusterV1alpha1().Gateways(c.namespace).Patch(context.TODO(), c.nodeConfig.Name, apitypes.MergePatchType, patch,
+			metav1.PatchOptions{})
+		return err
+	}); err != nil {
+		return fmt.Errorf("error when patching the Gateway with WireGuard information, error: %s", err)
+	}
+
+	return nil
+}
+
+// addWireGuardRoute initializes the WireGuard IP route on the host.
+// If the route already exists, it will do nothing.
+func (c *MCDefaultRouteController) addWireGuardRoute(remoteGatewayIP string) error {
+	_, remoteGatewayIPCIDR, err := net.ParseCIDR(remoteGatewayIP + "/32")
+	if err != nil {
+		return err
+	}
+	return c.routeClient.AddRouteForLink(remoteGatewayIPCIDR, c.wireGuardConfig.LinkIndex)
```

Comment:
I didn't see you provide logs in `AddRouteForLink`,maybe add a log line to indicate what's the route is here before this line? can be a level 2 log for future debugging.

Reply:
Added log.